### PR TITLE
Support GHC 8.10

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

**Comment by sjakobi** [comment]
CI with 8.10 failed due to https://github.com/conal/checkers/issues/44.

**Comment by sjakobi** [comment]
I have added an upper bound on `checkers` to work around the build failure.

Travis should succeed too now: https://travis-ci.org/github/hedgehogqa/haskell-hedgehog/builds/669930888

**Comment by moodmosaic** [comment]
@sjakobi, cool :beers: I just re-run the AppVeyor build.

**Comment by sjakobi** [comment]
Not sure what's up with AppVeyor…

Could you try deleting its cache?

**Comment by moodmosaic** [comment]
@sjakobi 

>Could you try deleting its cache?

There is an option "Save build cache in Pull Requests" which is turned off.

File: .travis.yml [hunk 1]
```diff
@@ -56,57 +59,31 @@ before_install:
   - TOP=$(pwd)
   - "HCNUMVER=$(${HC} --numeric-version|perl -ne '/^(\\d+)\\.(\\d+)\\.(\\d+)(\\.(\\d+))?$/; print(10000 * $1 + 100 * $2 + ($3 == 0 ? $5 != 1 : $3))')"
   - echo $HCNUMVER
-  - CABAL="$CABAL -vnormal+nowrap+markoutput"
+  - CABAL="$CABAL -vnormal+nowrap"
   - set -o pipefail
-  - |
```
Comment (moodmosaic) [inline .travis.yml:61]: Are these (and all other changes not directly related to GHC 8.10) warranted?

Comment (sjakobi) [reply .travis.yml:61]: Well I had to use a more recent version of `haskell-ci` that supports 8.10. These changes are due to the differences between that new version and the old version originally used to generate the file. There are probably some beneficial changes in there too…

**Comment by sjakobi** [comment]
> > Could you try deleting its cache?
> 
> There is an option "Save build cache in Pull Requests" which is turned off.

Hm, not sure what to do except try rerunning then.

**Comment by sjakobi** [comment]
The AppVeyor failures look similar to https://github.com/haskell/cabal/issues/4005. How about turning off builds with GHC 8.0?

**Comment by sjakobi** [comment]
OK. AppVeyor and Travis are finally both green now.

**Comment by jacobstanley** [comment]
@sjakobi did you use `bin/regenerate-travis.sh` for this? I think it's missing the `-Wall -Werror` that we used to have

**Comment by sjakobi** [comment]
> @sjakobi did you use `bin/regenerate-travis.sh` for this? I think it's missing the `-Wall -Werror` that we used to have

Oh no, I wasn't aware of that! It's strange that `--copy-fields=all` doesn't show up in the incantation in the .travis.yml.

Why don't you use the more conventional `cabal.hasell-ci` to configure `haskell-ci`?

**Comment by sjakobi** [comment]
[The Travis job with GHC 8.0](https://travis-ci.org/github/hedgehogqa/haskell-hedgehog/jobs/671113793) is failing now because of the combination of `-Werror=missing-methods` with `-Werror`. Previously [it just produced a warning](https://travis-ci.org/github/hedgehogqa/haskell-hedgehog/jobs/670602896#L752-L753).

I suspect it's a `haskell-ci` bug to pass the unsupported `-Werror=missing-methods` flag to a GHC < 8.2.

**Comment by sjakobi** [comment]
I reported the `haskell-ci` issue in https://github.com/haskell-CI/haskell-ci/issues/377.

**Comment by sjakobi** [comment]
I acted on @phadej's excellent advice in https://github.com/haskell-CI/haskell-ci/issues/377#issuecomment-609384926 and improved the `haskell-ci` incantation in a way that made the `regenerate-travis.sh` script unnecessary. In the future, a simple `haskell-ci regenerate` will be sufficient.

Unless https://github.com/conal/checkers/issues/44 gets merged very soon, I plan no further changes. ;)

File: bin/regenerate-travis.sh [hunk 2]
```diff
@@ -1,5 +0,0 @@
-#!/bin/sh -eu
-
-# .travis.yml is generated using the 'haskell-ci' available on Hackage
-
-haskell-ci regenerate --copy-fields=all
```
Comment (moodmosaic) [inline bin/regenerate-travis.sh:5]: Is `haskell-ci --copy-field=all cabal.project` an alternative? If yes, I think it makes sense to remove this (as you did).

Comment (sjakobi) [reply bin/regenerate-travis.sh:5]: Indeed. In general, once the correct incantation is recorded in the comment at the top of the `.travis.yml`, `haskell-ci regenerate` should be sufficient.

Comment (moodmosaic) [reply bin/regenerate-travis.sh:5]: Good to know :+1: Yes, as per @phadej in https://github.com/haskell-CI/haskell-ci/issues/377#issuecomment-609384926.

**Comment by jacobstanley** [comment]
> Why don't you use the more conventional cabal.hasell-ci to configure haskell-ci?

I wasn't aware, thanks for the changes! :tada:

**Comment by jacobstanley** [comment]
Note @erikd has updated the `template-haskell` bounds to `< 2.17` on the existing Hackage version `hedgehog-1.0.2`, so that should now be good to go also.

Thanks all :tada:

**Comment by sjakobi** [comment]
Thanks for the merge!

This was a bit of a yak shave! ;)

**Pull request merged**